keras/layers/recurrent.py

```diff
@@ -1238,31 +1243,56 @@ def build(self, input_shape):
             constraint=self.recurrent_constraint)
 
         if self.use_bias:
-            self.bias = self.add_weight(shape=(self.units * 3,),
+            if not self.reset_after:
+                bias_shape = (3 * self.units,)
+            else:
+                # separate biases for input and recurrent kernels
+                # Note: the shape is intentionally different from CuDNNGRU biases
+                # `(2 * 3 * self.units,)`, so that we can distinguish the classes
+                # when loading and converting saved weights.
+                bias_shape = (2, 3 * self.units)
+            self.bias = self.add_weight(shape=bias_shape,
                                         name='bias',
                                         initializer=self.bias_initializer,
                                         regularizer=self.bias_regularizer,
                                         constraint=self.bias_constraint)
+            if not self.reset_after:
+                self.input_bias, self.recurrent_bias = self.bias, None
+            else:
+                self.input_bias, self.recurrent_bias = self.bias[0], self.bias[1]
```

Comment:
The CNTK errors on Travis CI seem to come from this line. `self.bias[0]` results in a `(1, 3 * self.units)` 2-D tensor in CNTK.

Reply:
Aah. Thanks!

Reply:
So it means slicing in CNTK is not numpy compatible, right?

Reply:
Yes, the behavior is different from numpy.

Reply:
I filed a bug report https://github.com/Microsoft/CNTK/issues/2938. In the meanwhile we have to find a way how to slice it to 1D. API docs is not helpful.

Reply:
Ok, solution seems to be:`K.reshape(self.bias[0], (-1,))` to ensure 1D shape.